eng/scripts/Language-Settings.ps1

```diff
@@ -787,12 +387,81 @@ function Validate-java-DocMsPackages ($PackageInfo, $PackageInfos, $DocValidatio
+  # The install-rex-validation-tool.yml will install the java2docfx jar file into the Build.BinariesDirectory
+  # which is a DevOps variable for the directory. In PS that variable is BUILD_BINARIESDIRECTORY.
+  # The reason why this is necessary is that the command for java2docfx is in the following format:
+  # java –jar java2docfx-1.0.0.jar.jar --packagesJsonFile "C\temp\package.json"
+  # or
+  # java –jar java2docfx-1.0.0.jar --package "<GroupId>:<ArtifactId>:<Version>"
+  # which means we need to know where, exactly, because the java command requires the full path
+  # to the jar file as an argument
+  $java2docfxJar = $null
+  if (!$Env:BUILD_BINARIESDIRECTORY) {
+    LogError "Env:BUILD_BINARIESDIRECTORY is not set and this is where the java2docfx jar file should be installed."
+    return $false
+  }
+  $java2docfxDir = Join-Path $Env:BUILD_BINARIESDIRECTORY "java2docfx"
+  if (!(Test-Path $java2docfxDir)) {
+    LogError "There should be a java2docfx directory under Env:BUILD_BINARIESDIRECTORY. Ensure that the /eng/pipelines/templates/steps/install-rex-validation-tool.yml template was run prior to whatever step is running this."
+    return $false
+  }
+  $java2docfxJarLoc = @(Get-ChildItem -Path $java2docfxDir -File -Filter "java2docfx*.jar")
+  if (!$java2docfxJarLoc) {
+    LogError "The java2docfx jar file should be installed in $java2docfxDir and is not there."
     return $false
+  } else {
+    # In theory, this shouldn't happen as the install-rex-validation-tool.yml is the only thing
+    # that'll ever install the jar
+    if ($java2docfxJarLoc.Count -gt 1) {
+        Write-Host "There were $($java2docfxJarLoc.Count) java2docfx jar files found in $Build_BinariesDirectory, using the first one"
+    }
+    $java2docfxJar = $java2docfxJarLoc[0]
+    Write-Host "java2docfx jar location=$java2docfxJar"
+  }
+
+  $allSuccess = $true
+  $originLocation = Get-Location
+  foreach ($packageInfo in $PackageInfos) {
+    $artifact = "$($packageInfo.Group):$($packageInfo.Name):$($packageInfo.Version)"
+    $tempDirectory = Join-Path ([System.IO.Path]::GetTempPath()) "$($packageInfo.Group)-$($packageInfo.Name)-$($packageInfo.Version)"
+    New-Item $tempDirectory -ItemType Directory | Out-Null
+    # Set the location to the temp directory. The reason being is that it'll effectively be empty, no
+    # other jars, no POM files aka nothing Java related to pick up.
+    Set-Location $tempDirectory
+    try {
+      Write-Host "Calling java2docfx for $artifact"
+      Write-Host "java -jar ""$java2docfxJar"" -p ""$artifact"""
+      $java2docfxResults = java `
+      -jar "$java2docfxJar"`
+      -p "$artifact"
+      # JRS-TODO: The -o option is something I'm currently questioning the behavior of but
+      # I can do some initial testing without that option being set
+      # -p "$artifact" `
+      # -o "$tempDirectory"
+
+      if ($LASTEXITCODE -ne 0) {
+        LogWarning "java2docfx failed for $artifact"
```

Comment:
Should the failed artifacts also be collected and displayed at the end in the event of failure?

Reply:
@danieljurek unfortunately, no. Being that the very next line dumps the output from the java2docfx command, it won't actually be helpful to save them to the end. Also, [JS is doing the exact same thing as I am above](https://github.com/Azure/azure-sdk-for-js/blob/main/eng/scripts/Language-Settings.ps1#L444) except that it's not logging a warning and probably should since the colored text would be more helpful when looking at the logs than a write-host would be.